app/controllers/documents_controller.rb

```diff
@@ -253,8 +252,50 @@ def update
   end
 
   protected
-  def can_use_edit_token
-    if session[:lms_authenticated_user] || @organization[:enable_anonymous_actions] || has_role("designer")
+  def can_use_edit_token(lms_course_id = nil)
+    false
+    if @organization[:enable_anonymous_actions]
+      true
+    elsif has_role('designer')
+      true
+    elsif is_lms_authenticated_user? && has_canvas_access_token? && lms_course_id
+      if @document == nil
+        true
+      elsif lms_course_id == nil
+        true
+      elsif authorized_to_edit_course(lms_course_id)
+        true
+      end
+    end
```

Comment:
Don't we need a return of false at the end here? If they aren't logged in, but there is a course_id